core/src/main/scala/org/http4s/Header.scala

```diff
@@ -141,56 +142,93 @@ object Header {
 
     /** Transform this header into a [[Header.Raw]]
       */
-    def toRaw(a: A): Header.Raw
+    def toRawOne(a: A): Header.Raw
+
+    /** Transform this (potentially repeating) header into a [[Header.Raw]] */
+    def toRaw(a: F[A]): Header.Raw
```

Comment:
I'm a bit lost on this one.  Is this something rho needed?  Can you point me to it?

Reply:
https://github.com/http4s/rho/blob/c5a90fcebaee2524d5898969f38ec183b2eb8c86/core/src/main/scala/org/http4s/rho/RhoDslHeaderExtractors.scala#L146

I debated about just changing the existing `toRaw` rather than having a separate `toRawOne`, since most of the time you have a header with a semigroup, rather than a list of something, (or at least that seems to be the http4s pattern) where you might want to just do 1 of.

Reply:
Have any suggestions on this @rossabaker ?  Would it be better to have just `toRaw` and not a `toRawOne`?

Reply:
@rossabaker ^?

Reply:
`toRawOne` is equivalent to `toRaw` for single headers, and for headers with a Semigroup instance?  Only recurring headers without a Semigroup need this new signature?  And that is in practice only `Set-Cookie`?

Reply:
I don't think `toRaw(a: F[A]): Header.Raw` is sensible for `Set-Cookie`, because a `NonEmptyList` of `Set-Cookie` can't be reduced to a single header.

Reply:
What would be correct. I'm fine with dropping the `toRawOne`, but Rho will need the new version of `toRaw` that takes a `F[A]` instead of just an `A` in order for the types to work out.

Reply:
I guess what I'm saying is that there's no correct implementation of that signature for `Set-Cookie`, because you can't just concatenate the values of the `NonEmptyList` with a comma.  The return type would need to be a collection for that header.  It's the exception to all the other rules.

Reply:
Sooo.... are you saying we keep both signatures?  Do we just need better names?

Reply:
I'm concerned about what the need for this signature is telling us, because the one that joins with a comma is an invalid implementation.  Perhaps I need to spend more time studying that Rho link.

Reply:
`cats-effect-3` branch, right?  I might need to get hands on with this.  The types are more than I can contemplate on GitHub.

Reply:
Correct.  The important part for that piece of code is we have an `F[A]` where `toRaw` requires and `A`.   I can probably work around this, but it just seemed logical at the time that `toRaw` should be able to convert to and from raw in the `F[A]` form.

Reply:
Yeah, it's just that one exceptional header that makes that not work.  I'm playing type tetris with a local checkout now.

Reply:
Okay, https://github.com/http4s/rho/pull/538 compiles after my PR to this PR.  I don't know rho nearly as well as you, so I don't know whether it's acceptable, but maybe that's our way out of this?

Reply:
I think it'll be fine.  Thanks for looking into this!